File: frontend/upgrade_user.js
const admin = require('firebase-admin');
const path = require('path');
// Put the file in the same directory for simplicity
const serviceAccount = require('./service-account-key.json');

if (!admin.apps.length) {
  admin.initializeApp({
    credential: admin.credential.cert(serviceAccount)
  });
}

const db = admin.firestore();

async function upgrade() {
  const email = "taunhealy@gmail.com";
  console.log(`🧬 Upgrading ${email} to PRO...`);
  
  await db.collection('users').doc(email).set({
    tier: 'Pro',
    subscription_status: 'active',
    updated_at: admin.firestore.FieldValue.serverTimestamp()
  }, { merge: true });
  
  console.log("✅ Success: taunhealy@gmail.com is now a PRO member.");
}

upgrade().catch(err => {
  console.error("❌ Error:", err);
  process.exit(1);
});

File: monitor/upgrade_user.py
import firebase_admin
from firebase_admin import credentials, firestore
import os

def upgrade_user():
    # Use existing credentials path from monitor logic
    cred_path = r"k:\Kea\sole-node\monitor\service-account-key.json"
    if not os.path.exists(cred_path):
        print("❌ Error: Service account key not found.")
        return

    cred = credentials.Certificate(cred_path)
    if not firebase_admin._apps:
        firebase_admin.initialize_app(cred)

    db = firestore.client()
    user_ref = db.collection("users").document("taunhealy@gmail.com")
    
    user_ref.set({
        "tier": "Pro",
        "subscription_status": "active",
        "updated_at": firestore.SERVER_TIMESTAMP
    }, merge=True)
    
    print("✅ Success: taunhealy@gmail.com UPGRADED TO PRO.")

if __name__ == "__main__":
    upgrade_user()
